client/common/usePrevious.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * Used in Menubar to store the previous value of a number.
+ * @param value - The current value to track.
+ * @returns The previous value before the current render, or undefined if none.
+ */
+export default function usePrevious(value: number): number | undefined {
```

Comment:
is there any way we can avoid having it be typed to numbers specifically? Maybe this is a rare case where it makes sense to use `any` or `unknown`? that way the hook can be used for more general cases in the future